python/cudf/cudf/core/frame.py

```diff
@@ -1248,14 +1251,26 @@ def replace(self, to_replace, replacement):
                         column=col,
                     )
 
-                    copy_data[name] = col.find_and_replace(
-                        col_to_replace, col_replacement, col_all_nan
-                    )
+                    # Index values to be replaced do not store in column
+                    # data but instead *_values*. Column accessor bypassed
+                    # to direct to Index.find_and_replace()
+                    if is_index:
+                        copy_data[name] = self.find_and_replace(
+                            to_replace=to_replace, replacement=replacement,
+                        )
+                    else:
+                        copy_data[name] = col.find_and_replace(
+                            col_to_replace, col_replacement, col_all_nan
+                        )
                 except KeyError:
                     # Do not change the copy_data[name]
                     pass
 
-        result = self._from_table(Frame(copy_data, self.index))
+            if is_index:
+                result = self._from_table(Frame(copy_data))
+            else:
+                result = self._from_table(Frame(copy_data, self.index))
```

Comment:
We can use the internal member of `._index` which is `None` in the case of `Index` objects.
```suggestion
            result = self._from_table(Frame(copy_data, self._index))
```